CMakeLists.txt

```diff
@@ -33,10 +35,9 @@ option(MICM_BUILD_DOCS "Build the documentation" OFF)
 option(MICM_ENABLE_LLVM "Build with LLVM support for JIT-compiling" OFF)
 option(MICM_ENABLE_TESTS "Build the tests" ON)
 option(MICM_ENABLE_PROFILE "Profile MICM Solver" OFF)
-set(MICM_GPU_TYPE "None" CACHE STRING "The GPU type being targeted")
+option(MICM_ENABLE_CUDA "Enable CUDA support" OFF)
```

Comment:
This line seems redundant. When a user sets `MICM_GPU_TYPE`, it implies that CUDA is enabled. On the other hand, when `MICM_ENABLE_CUDA` option is enabled, a user still needs to specify what type of NVIDIA GPUs they want to compile.

Reply:
That's true. But we also end up setting this option later. Since we will eventually compile for all GPU types, I'm wondering if our default to should be to target all GPUs if there's no shock to performance. I can revert this though

Reply:
If that is the case, I think we can just set `MICM_GPU_TYPE` to something like `all-major`.

I do not think that compiling for all GPU types will hurt the performance (maybe missing some architecture-specific optimization), but the binary file size may be larger and the compilation time may be longer. I still think it is the user's responsibility to tell MUSICA/MICM what type of GPUs are you going to use. In addition, if we plan to run on AMD or Intel GPUs in the future, the `MICM_ENABLE_CUDA` option won't work but the `MICM_GPU_TYPE` can still be used to point to the right libraries/compilers.